packages/reputation-miner/ReputationMiner.js

```diff
@@ -173,7 +173,12 @@ class ReputationMiner {
    */
   getAdjacentKey(key) {
     const sortedHashes = Object.keys(this.reverseReputationHashLookup).sort();
-    const keyPosition = sortedHashes.indexOf(soliditySha3(key));
+    let keyPosition = sortedHashes.indexOf(soliditySha3(key));
+    if (keyPosition === -1) {
+      sortedHashes.push(soliditySha3(key))
+      sortedHashes.sort()
+      keyPosition = sortedHashes.indexOf(soliditySha3(key));
+    }
```

Comment:
No, this is required for the good clients. When we're using an adjacent reputation to prove that something doesn't exist in the tree, we insert the key in to the `sortedHashes` array to work out the reputation that would be adjacent if it did exist.

This wasn't an issue before, because we were explicitly adding it to the mapping [here](https://github.com/JoinColony/colonyNetwork/blob/88bddea3aa33a5e86ac057e39354a6341011012f/packages/reputation-miner/ReputationMiner.js#L352) just before it was really in the tree, and then calling this function. But because now it might be many more updates before a reputation we're asking the adjacent reputation of is actually in the tree, we don't want to confuse any interim `getAdjacentKey` calls before it's actually in the tree, we can't add it. So we insert it here.